lib/emitter/src/Events/EnrolmentCreated.php

```diff
@@ -37,7 +37,7 @@ public function read(array $opts) {
             ],
             'object' => $this->read_course($opts),
             'context' => [
-                'instructor' => $this->read_user($opts, 'instructor'),
+                'instructor' => parent::read($opts)['actor'],
```

Comment:
We should probably use `$this->read_user($opts, 'user')`, but I'm not sure this is the correct fix. I think actually the instructor needs to provided from the translator and expander. The reason issue #121 exists, is probably in the case where either:
- the user enrols themselves rather than being enrolled by a instructor, hence there's no instructor passed through
- the instructor enrols a user, but the user that's passed from the expander and translator is the instructor instead of the user being enrolled, which has also resulted in the instructor not being set

Reply:
Hi @ryansmith94,

Thanks for your feedback. I agree with you, we need the instructor there. The problem is in EnrolmentCreated.php in translator. It didn't assign 'instructor_email' or 'user_email'. I changed it like this (added both instructor_email and user_email),
 public function read(array $opts) {
        return [array_merge(parent::read($opts)[0], [
            'recipe' => 'enrolment_created',
            'user_id' => $opts['relateduser']->id,
            'user_url' => $opts['relateduser']->url,
            'user_name' => $opts['relateduser']->fullname,
            'user_email' => $opts['relateduser']->email,
            'instructor_id' => $opts['user']->id,
            'instructor_url' => $opts['user']->url,
            'instructor_email' => $opts['user']->email,
            'instructor_name' => $opts['user']->fullname,
        ])];
    }
After I changed this and revert my previous changes in 'lib/emitter/src/Events/EnrolmentCreated.php' file, it worked properly for both self enrollments and also instructor enrollments. Thank you so much for this details reply and put me on to the correct track. I updated my pull request with latest changes. 
Dasu

Reply:
Hey @Dasuntha, you're very welcome, thanks for your reply. I see, good work tracking that down. That's great! You're very welcome. Thanks for updating your PR, I'll get this merged and released today.